build_tools/circle/build_doc.sh

```diff
@@ -109,7 +109,7 @@ conda update --yes --quiet conda
 conda create -n $CONDA_ENV_NAME --yes --quiet python="$PYTHON_VERSION" numpy scipy \
```

Comment:
so shouldn't we incorporate doc.txt here now?

Reply:
Modifying `build_tools/circle/build_doc.sh` to have 

```bash
# Configure the conda environment using provided versions
# Note: sphinx-gallery needs to be downloaded from conda-forge
conda config --append channels conda-forge
conda create -n $CONDA_ENV_NAME --yes --quiet python="${PYTHON_VERSION:-*}" cython \
  --file requirements/default.txt --file requirements/docs.txt \
  --file requirements/test.txt
conda install -n $CONDA_ENV_NAME --yes --quiet matplotlib="${MATPLOTLIB_VERSION:-*}"

source activate $CONDA_ENV_NAME
# Use numpydoc master (for now)
pip install git+https://github.com/numpy/numpydoc
```

will install all the needed dependencies from `requirements`. Having the separate line to install a specific version of matplotlib isn't elegant, but it works.